src/pages/ItemList.js

```diff
@@ -45,67 +45,93 @@ function ItemList({ token }) {
     fetchDocs(token);
   }, [token]);
 
-  useEffect(() => {
-    // Update "now" every second to force the page to re-render and keep the checkbox state up to date;
-    const intervalId = setInterval(() => {
-      setNow(Date.now());
-    }, ONE_SECOND);
-    return () => clearInterval(intervalId);
-  }, []);
-
   const isClicked = async ({ target: { checked, id } }) => {
     /**
      * Return early if the checkbox was already checked
      * because we don't want to purchase an item twice.
      */
+
+    // we obtain the object with the associated id WORKING
+    const itemId = docs.find((doc) => {
+      return doc.id === id;
+    });
+
+    const totalPurchases = itemId.total_purchases + 1;
+
+    const previousEstimate =
+      itemId.previous_estimate || itemId.purchase_interval;
+
+    const today = Date.now();
+
+    // gets the last purchased date for the item
+    const lastPurchased = itemId.last_purchased_date || today;
+
+    const daysSincePurchase = today - lastPurchased;
+
+    //msToDay function below
+    const daysSinceLastTransaction = Math.round(msToDay(daysSincePurchase));
+
     if (!checked) return;
```

Comment:
I think we may want to move `if(!checked) return;` back up towards the comment at the beginning of the function to which it's associated.